engine/engine.go

```diff
@@ -350,6 +362,7 @@ var _ ExplainableQuery = &compatibilityQuery{}
 type Query struct {
 	exec model.VectorOperator
 	opts *promql.QueryOpts
+	obs  model.ObservableVectorOperator
```

Comment:
Why don't we just use exec here?

Reply:
Hmm, but then how can we Analyze the query, because the Analyze method is only a part of ObservableVectorOperator interface

Reply:
We can typecast it later on?